<launch>

  <node pkg="hector_imu_attitude_to_tf" type="imu_attitude_to_tf_node" name="imu_attitude_to_tf_node" output="screen">
    <remap from="imu_topic" to="ano_imu" />
    <param name="base_stabilized_frame" type="string" value="imu_link" />
    <param name="base_frame" type="string" value="imu_link" />
  </node>

  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
  <!-- Frame names -->
  <param name="pub_map_odom_transform" value="true"/>
  <param name="map_frame" value="map" />
  <param name="base_frame" value="laser_frame" />
  <param name="odom_frame" value="laser_frame" />


  <!-- Tf use -->
  <param name="use_tf_scan_transformation" value="true"/>
  <param name="use_tf_pose_start_estimate" value="false"/>


  <!-- Map size / start point -->
  <param name="map_resolution" value="0.05"/>
  <param name="map_size" value="2048"/>
  <param name="map_start_x" value="0.5"/>
  <param name="map_start_y" value="0.5" />
  <param name="laser_z_min_value" value = "-1.0" />
  <param name="laser_z_max_value" value = "1.0" />
  <param name="map_multi_res_levels" value="2" />


  <param name="map_pub_period" value="1" />
  <param name="laser_min_dist" value="0.25" />
  <param name="laser_max_dist" value="11.5" />
  <param name="output_timing" value="false" />
  <param name="pub_map_scanmatch_transform" value="true" />
  <!--<param name="tf_map_scanmatch_transform_frame_name" value="scanmatcher_frame" />-->


  <!-- Map update parameters -->
  <param name="update_factor_free" value="0.4"/>
  <param name="update_factor_occupied" value="0.7" />    
  <param name="map_update_distance_thresh" value="0.05"/>
  <param name="map_update_angle_thresh" value="0.001" />


  <!-- Advertising config --> 
  <param name="advertise_map_service" value="true"/>
  <param name="scan_subscriber_queue_size" value="5"/>
  <param name="scan_topic" value="scan"/>
  </node>

  <!--
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster" args="0 0 0.1 -1.5707963 0 0 /base_link /laser_frame 50"/>

  node pkg="tf" type="static_transform_publisher" name="base_to_laser_broadcaster" args="0 0 0.1 -1.5707963 0 0 /base_stabilized /laser_frame 50"/

  <node pkg="tf" type="static_transform_publisher" name="map_to_odom_broadcaster" args="0 0 0 0 0 0 /map /odom 50"/>

-->

  <node pkg="rviz" type="rviz" name="rviz"
    args="-d $(find hector_slam_launch)/rviz_cfg/mapping_demo.rviz"/>

</launch> 
